testsuite/smoke-test/src/state_sync.rs

```diff
@@ -731,46 +755,111 @@ async fn test_validator_sync(
+/// If `fast_sync` is true, then the validator will use fast (snapshot)
+/// syncing. Otherwise, it will use the default syncing method.
+async fn test_validator_sync_and_participate(fast_sync: bool, epoch_changes: bool) {
+    // Create a swarm of 4 validators
+    let num_validators = 4;
+    let mut swarm = SwarmBuilder::new_local(num_validators)
         .with_aptos()
-        .with_init_config(Arc::new(|_, config, _| {
-            config.state_sync.state_sync_driver.bootstrapping_mode =
-                BootstrappingMode::DownloadLatestStates;
-            config.state_sync.state_sync_driver.continuous_syncing_mode =
-                ContinuousSyncingMode::ApplyTransactionOutputs;
+        .with_init_config(Arc::new(move |_, config, _| {
+            if fast_sync {
+                // Set the bootstrapping mode to fast syncing
+                config.state_sync.state_sync_driver.bootstrapping_mode =
+                    BootstrappingMode::DownloadLatestStates;
+                config.state_sync.storage_service.max_state_chunk_size = 30;
+            }
+        }))
+        .with_init_genesis_config(Arc::new(|genesis_config| {
+            // Shorten the required proposer history to speed up the test
+            let consensus_config = match genesis_config.consensus_config.clone() {
+                OnChainConsensusConfig::V1(consensus_config) => consensus_config,
+                OnChainConsensusConfig::V2(consensus_config) => consensus_config,
+                config => unimplemented!(
+                    "This test requires a V1 or V2 consensus config, but got: {:?}",
+                    config
+                ),
+            };
+            let leader_reputation_type = match &consensus_config.proposer_election_type {
+                ProposerElectionType::LeaderReputation(leader_reputation_type) => {
+                    leader_reputation_type
+                },
+                proposer_election_type => panic!(
+                    "This test requires a leader reputation proposer election, but got: {:?}",
+                    proposer_election_type
+                ),
+            };
+            let proposer_and_voter_config = match &leader_reputation_type {
+                LeaderReputationType::ProposerAndVoterV2(proposer_and_voter_config) => {
+                    proposer_and_voter_config
+                },
+                leader_reputation_type => panic!(
+                    "This test requires a proposer and voter V2 leader reputation, but got: {:?}",
+                    leader_reputation_type
+                ),
+            };
+            genesis_config.consensus_config = OnChainConsensusConfig::V1(ConsensusConfigV1 {
+                proposer_election_type: ProposerElectionType::LeaderReputation(
+                    LeaderReputationType::ProposerAndVoter(ProposerAndVoterConfig {
+                        proposer_window_num_validators_multiplier: 0,
```

Comment:
hmm, never used these as 0 before :D


I would've put all 3 to 1, and that would be the window of 4 rounds (i.e. 4 validators). probably more realistic to test it can recover from no history (vs 0 where it immediately succeeds, and there is nothing to recover from)

Reply:
Gotcha, SGTM. 😄 Done, thanks Igor!